### PR TITLE
Doc - Add warning note about persistence message deletion

### PR DESCRIPTION
**Pull request by richardimaoka**
Refs #18886

image

### CODE REVIEW

File: akka-docs/rst/java/persistence.rst [hunk 1]
```diff
@@ -385,6 +385,10 @@ Deleting messages in event sourcing based applications is typically either not u
 up until the sequence number of the data held by that snapshot can be issued to safely delete the previous events
 while still having access to the accumulated state during replays - by loading the snapshot.
 
+.. warning::
+  If you are using :ref:`persistence-query-scala`, query results will be missing deleted messages in a journal.
```
Comment (johanandren) [inline akka-docs/rst/java/persistence.rst]: Ref should go to `persistence-query-java`

**Comment by johanandren** [comment]
LGTM except Java link pointing to Scala docs

File: akka-docs/rst/java/persistence.rst (hunk 1 above)
Comment (richard-imaoka) [inline akka-docs/rst/java/persistence.rst]: Oops, thanks for finding it out. I fixed this in thew new commit.

File: akka-docs/rst/java/persistence.rst [hunk 2]
```diff
@@ -385,6 +385,10 @@ Deleting messages in event sourcing based applications is typically either not u
 up until the sequence number of the data held by that snapshot can be issued to safely delete the previous events
 while still having access to the accumulated state during replays - by loading the snapshot.
 
+.. warning::
+  If you are using :ref:`persistence-query-java`, query results will be missing deleted messages in a journal.
```
Comment (patriknw) [inline akka-docs/rst/java/persistence.rst]: it depends on the journal implementation, I think the leveldb implementation is a completely detached materialized view. Reword to "may be missing deleted messages in a journal, depending on how deletions are implemented in the journal plugin"

**Review by patriknw — CHANGES_REQUESTED** [review]

File: akka-docs/rst/java/persistence.rst (hunk 2 above)
Comment (richard-imaoka) [inline akka-docs/rst/java/persistence.rst]: Thanks for the feedback. I modified the warning text - could you see if that looks ok?

**Comment by richard-imaoka** [comment]
Hi, I responded to the review comments. Appreciate it if you can take a look when you have time.
Thanks

**Review by patriknw — APPROVED** [review]

**Comment by patriknw** [comment]
LGTM

**Comment by ktoso** [comment]
LGTM, thanks

**Pull request merged**